apidocs: Remove template of get-events.

MSurfer20 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing plan:** <!-- How have you tested? -->


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

MSurfer20 commented:
Attaching the diffs:
```
old-get-events new-get-events
233c233
< <div class="codehilite"><pre><span></span><code><span class="ch">#!/usr/bin/env python</span>
---
> <div class="codehilite" data-code-language="Python"><pre><span></span><code><span class="ch">#!/usr/bin/env python3</span>
239c239
< <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">&quot;~/zuliprc&quot;</span><span class="p">)</span>
---
> <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">"~/zuliprc"</span><span class="p">)</span>
244,247c244,247
< <span class="nb">print</span><span class="p">(</span><span class="n">client</span><span class="o">.</span><span class="n">get_events</span><span class="p">(</span>
<     <span class="n">queue_id</span><span class="o">=</span><span class="s2">&quot;1515010080:4&quot;</span><span class="p">,</span>
<     <span class="n">last_event_id</span><span class="o">=-</span><span class="mi">1</span>
< <span class="p">))</span>
---
> <span class="nb">print</span><span class="p">(</span><span class="n">client</span><span class="o">.</span><span class="n">get_events</span><span class="p">(</span><span class="n">queue_id</span><span class="o">=</span><span class="s2">"1515010080:4"</span><span class="p">,</span> <span class="n">last_event_id</span><span class="o">=-</span><span class="mi">1</span><span class="p">))</span>
> <span class="c1"># `call_on_each_message` and `call_on_each_event` will automatically register</span>
> <span class="c1"># a queue for you.</span>
> <span class="nb">print</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
249,250d248
< <p><code>call_on_each_message</code> and <code>call_on_each_event</code> will automatically register
< a queue for you.</p>
319a318,322
> <p><strong>Note</strong>: The parameters documented above are optional in the sense that
> even if you haven't registered a queue by explicitly requesting the
> <code>http://localhost:9991/api/v1/register</code> endpoint, you could pass the parameters for
> <a href="/api/register-queue">the <code>http://localhost:9991/api/v1/register</code> endpoint</a> to this
> endpoint and a queue would be registered in the absence of a <code>queue_id</code>.</p>
old-get-events new-get-events
```

MSurfer20 commented:
@timabbott I realised that the `events:get` python example was hardcoded not because of the `sys` configuration, but because it's no possible to test the code examples for it with our current backend suite. Should I let this template be as well?

timabbott commented:
I think it should be possible to test `events:get`; one just needs to call `events:register` first to get a `queue_id`.

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -165,6 +165,7 @@ paths:
         `{{ api_url}}/v1/register` endpoint, you could pass the parameters for
         [the `{{ api_url}}/v1/register` endpoint](/api/register-queue) to this
         endpoint and a queue would be registered in the absence of a `queue_id`.
+      x-requires-sys: true
```
maybe `x-python-examples-extra-imports: ["sys"]` would be a cleaner interface?

MSurfer20 replied:
That seems more robust. Changed to such interface.

timabbott commented on zerver/openapi/python_examples.py:
```diff
@@ -1097,6 +1097,20 @@ def deregister_queue(client: Client, queue_id: str) -> None:
+
+    # {code_example|start}
+    # `call_on_each_message` and `call_on_each_event` will automatically register
+    # a queue for you.
```
This feels like a non sequitur with the new placement.  I think this belongs in the description, not the code examples.

MSurfer20 replied:
Seems right. Moved to a Note in description.

MSurfer20 commented:
> I think it should be possible to test `events:get`; one just needs to call `events:register` first to get a `queue_id`.

@timabbott Hmmm I felt that because of the comment block:
```
    # Note that the example for api/get-events is not tested.
    # Since, methods such as client.get_events() or client.call_on_each_message
    # are blocking calls and since the event queue backend is already
    # thoroughly tested in zerver/tests/test_event_queue.py, it is not worth
    # the effort to come up with asynchronous logic for testing those here.
```

MSurfer20 commented:
@timabbott This is not passing tests, because the event being retuned is `heartbeat`, which isn't present in the OpenAPI data. Is it not being documented intentional?

timabbott commented:
That's a bug!  `heartbeat` is an event that's sent every minute on an idle connection basically to defeat buggy networking equipment that kills connections with no traffic for a minute.

Presumably the reason this is undocumented is that clients don't need to do anything when they get such an event, and thus we have no `test_events.py` tests to exercise it.

timabbott commented:
I've pushed a commit that documents `heartbeat` events, since most of the work is writing documentation for them: 64aa8f80a06c67696202936eccc483643ae9bafa

We talk about these in a few places in the API documentation, and it'd be nice to cross-link them, but we don't have the anchor support do to that yet.  So mostly mentioning this as a reminder of something to do as a follow up.

MSurfer20 commented:
@timabbott  Thanks for the fix, after which the tests are now passing :). Addressed the comments above, and attaching the diffs of the files:
```
old-get-events new-get-events
223a224,225
> <p><strong>Note</strong>: <code>call_on_each_message</code> and <code>call_on_each_event</code> will automatically register
> a queue for you.</p>
233c235
< <div class="codehilite"><pre><span></span><code><span class="ch">#!/usr/bin/env python</span>
---
> <div class="codehilite" data-code-language="Python"><pre><span></span><code><span class="ch">#!/usr/bin/env python3</span>
239c241
< <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">&quot;~/zuliprc&quot;</span><span class="p">)</span>
---
> <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">"~/zuliprc"</span><span class="p">)</span>
244,247c246,247
< <span class="nb">print</span><span class="p">(</span><span class="n">client</span><span class="o">.</span><span class="n">get_events</span><span class="p">(</span>
<     <span class="n">queue_id</span><span class="o">=</span><span class="s2">&quot;1515010080:4&quot;</span><span class="p">,</span>
<     <span class="n">last_event_id</span><span class="o">=-</span><span class="mi">1</span>
< <span class="p">))</span>
---
> <span class="n">result</span> <span class="o">=</span> <span class="n">client</span><span class="o">.</span><span class="n">get_events</span><span class="p">(</span><span class="n">queue_id</span><span class="o">=</span><span class="n">queue_id</span><span class="p">,</span> <span class="n">last_event_id</span><span class="o">=-</span><span class="mi">1</span><span class="p">)</span>
> <span class="nb">print</span><span class="p">(</span><span class="n">result</span><span class="p">)</span>
249,250d248
< <p><code>call_on_each_message</code> and <code>call_on_each_event</code> will automatically register
< a queue for you.</p>
319a318,322
> <p><strong>Note</strong>: The parameters documented above are optional in the sense that
> even if you haven't registered a queue by explicitly requesting the
> <code>http://localhost:9991/api/v1/register</code> endpoint, you could pass the parameters for
> <a href="/api/register-queue">the <code>http://localhost:9991/api/v1/register</code> endpoint</a> to this
> endpoint and a queue would be registered in the absence of a <code>queue_id</code>.</p>
old-get-events new-get-events
```

MSurfer20 commented:
@timabbott A gentle ping regarding this.

timabbott commented:
Merged, after making the editorial decision to remove the `call_on_each_events` note, rather than moving it.  Thanks @MSurfer20!
